app/services/refresh_materialized_views.rb

```diff
@@ -39,37 +42,44 @@ def self.tz
 
   delegate :tz, :set_last_updated_at, to: self
 
-  def refresh
-    # LatestBloodPressuresPerPatientPerMonth should be refreshed before
-    # LatestBloodPressuresPerPatientPerQuarter and LatestBloodPressuresPerPatient
+  V1_MATVIEWS = %w[
+    LatestBloodPressuresPerPatientPerMonth
+    LatestBloodPressuresPerPatient
+    LatestBloodPressuresPerPatientPerQuarter
+    BloodPressuresPerFacilityPerDay
+    PatientRegistrationsPerDayPerFacility
+    MaterializedPatientSummary
+  ].freeze
+  V2_MATVIEWS = %w[
+    ReportingPipeline::PatientBloodPressuresPerMonth
+    ReportingPipeline::PatientStatesPerMonth
+    ReportingPipeline::PatientVisitsPerMonth
+  ].freeze
```

Comment:
@rsanheim We would only need:

```ruby
ReportingPipeline::PatientStatesPerMonth.refresh
```

The `cascade: true` in the scenic refresh function takes care of refreshing the dependent views.

Reply:
Cascade [doesn't work](https://github.com/scenic-views/scenic/issues/275), and it relies some [own SQL parsing and regexp](https://github.com/scenic-views/scenic/blob/3600a485797fe1dbf30d152cc60b6d2318f04c48/lib/scenic/adapters/postgres/refresh_dependencies.rb#L29) that I wouldn't really trust even if it did work for our case 😄 .  I'll make sure to remove all the `cascade: true` options for the new matviews.

I think its fine to know what our dependency chain is and declare it explicitly in the refresh object...its a pretty key part of these matviews and we shouldn't be refreshing them in isolation for most cases anyways.  New tests that rely on the matviews can call `refresh_v2` for their setup, the same way we had a repeated `refresh` call for the old ones.

Reply:
Oh well. Thanks for catching that btw! 😅